Source/Diagnostics/MultiDiagnostics.cpp

```diff
@@ -22,11 +22,13 @@ MultiDiagnostics::MultiDiagnostics ()
     alldiags.resize( ndiags );
     for (int i=0; i<ndiags; i++){
         if ( diags_types[i] == DiagTypes::Full ){
-            alldiags[i] = std::make_unique<FullDiagnostics>(i, diags_names[i]);
+            alldiags[i] = std::make_unique<FullDiagnostics>(i, diags_names[i], diags_types[i]);
+        } else if ( diags_types[i] == DiagTypes::TimeAveraged ){
+            alldiags[i] = std::make_unique<FullDiagnostics>(i, diags_names[i], diags_types[i]);
```

Comment:
Should this be `TimeAveragedDiagnostics` instead? I think this should fix the clang-tidy error "repeated branch in conditional chain [bugprone-branch-clone,-warnings-as-errors]":
```suggestion
            alldiags[i] = std::make_unique<TimeAveragedDiagnostics>(i, diags_names[i], diags_types[i]);
```

Reply:
We don't actually have a full class `TimeAveragedDiagnostics` because our diagnostics are basically `FullDiagnostics` with the addition of time averaging and without particles. 

I am already having new changes ready to address the clang-tidy errors.
What I was thinking was to just do
```C++
if ( diags_types[i] == DiagTypes::Full || diags_types[i] == DiagTypes::TimeAveraged ){
    alldiags[i] = std::make_unique<FullDiagnostics>(i, diags_names[i], diags_types[i]);
```